java/pom.xml

```diff
@@ -358,6 +358,19 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-source-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>attach-sources</id>
+            <phase>package</phase>
+            <goals>
+              <goal>jar-no-fork</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
```

Comment:
What about [maven-dependency-plugin](https://maven.apache.org/plugins/maven-dependency-plugin/) it tells Maven to resolve all dependencies and their source attachments with `mvn dependency:sources --debug`?

```
[INFO] --- maven-dependency-plugin:2.8:sources (default-cli) @ wks-reproduce ---
Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/arrow/arrow-vector/9.0.0/arrow-vector-9.0.0-sources.jar (946 kB at 5.1 MB/s)
```

Reply:
For release version 9.0.0, it works well. How can I  get the source for the current branch master, I compiled 10.0.0-SNAPSHOT by myself.